sourcegraph/sourcegraph
# web: integrate OpenTelemetry client observability packages

## Context 

Add basic configuration of the OpenTelemetry tracing to the web application. [Based on prior work](https://github.com/sourcegraph/sourcegraph/pull/37843). 
[Honeycomb guide](https://docs.honeycomb.io/getting-data-in/opentelemetry/browser-js/). 

## Changes

1. Removed `ENABLE_MONITORING` env variable in favor of `ENABLE_SENTRY` and `ENABLE_OPEN_TELEMETRY`. Required for `standalone-web-server` in the development environment to avoid accidentally sending telemetry data from the local env.
2. Integrated `@opentelemetry` packages and moved them into a separate chunk. Added new script tags required for the new initial chunk.

    <img width="840" alt="Screen Shot 2022-07-28 at 13 03 46" src="https://user-images.githubusercontent.com/3846380/181424752-317865c0-712a-4235-87c8-fc29d3f1ceb8.png">

3. Enabled [@opentelemetry/instrumentation-fetch](https://github.com/open-telemetry/opentelemetry-js/tree/main/experimental/packages/opentelemetry-instrumentation-fetch).

    <img width="1156" alt="image" src="https://user-images.githubusercontent.com/23356519/181091983-a73fb6a9-5b12-4219-8cc9-1779b785d5c8.png">

**Note**: we won't be collecting data until [the backend tunnel PR](https://github.com/sourcegraph/sourcegraph/pull/39453) is merged. 

Closes https://github.com/sourcegraph/sourcegraph/issues/39273
Closes https://github.com/sourcegraph/sourcegraph/issues/39281

## Test plan

1. Checkout to https://github.com/sourcegraph/sourcegraph/pull/39453
2. Set `dev-private` site config to use `opentelemetry` 
3. Set `observability.client` in the site config.
4. `sg start otel`
5. `sg start` 
6. Open a web page with API requests.
7. See traces logged to Jaeger.
